Assets/Scripts/SS3D/Core/Behaviours/Actor.cs

```diff
@@ -1,187 +1,270 @@
-        public Matrix4x4 WorldToLocalMatrix => TransformCache.worldToLocalMatrix;
-        #endregion
+        protected void OnDisable()
+        {
+            OnDisabled();
+        }
 
-        #region SETUP
-        private void Awake()
+        protected void Awake()
         {
             Initialize();
+
             OnAwake();
         }
 
-        private void Start()
+        protected void Start()
         {
-            AddEventListeners();
             OnStart();
         }
 
-        private void OnDestroy()
+        protected void OnDestroy()
         {
+            RemoveEventListeners();
+
             OnDestroyed();
         }
 
+        /// <summary>
+        /// Called once the script is loaded and registered on ActorLocator.
+        /// </summary>
+        protected virtual void OnAwake() { }
+
+        /// <summary>
+        /// Called once the start function is called.
+        /// </summary>
+        protected virtual void OnStart() { }
+
+        /// <summary>
+        /// Called once the Actor is destroyed, after removing the Actor from the ActorLocator and all event listeners.
+        /// </summary>
+        protected virtual void OnDestroyed() { }
+
+        /// <summary>
+        /// Called when the Actor's GameObject is enabled. 
+        /// </summary>
+        protected virtual void OnEnabled() { }
+
+        /// <summary>
+        /// Called when the Actor's GameObject is disabled.
+        /// </summary>
+        protected virtual void OnDisabled() { }
+
+        /// <summary>
+        /// Initializes the Actor, caching the Transform and the GameObject.
+        /// </summary>
         private void Initialize()
         {
-            TransformCache = transform;
-            GameObjectCache = gameObject;
+            _transformCache = transform;
+            _gameObjectCache = gameObject;
 
             _initialized = true;
         }
 
-        private void AddEventListeners()
-        {
-            _eventHandles.Add(LastPreUpdateEvent.AddListener(OnPreUpdate));
-            _eventHandles.Add(UpdateEvent.AddListener(OnUpdate));
-            _eventHandles.Add(LateUpdateEvent.AddListener(OnLateUpdate));
-        }
-
+        /// <summary>
+        /// Removes all subscribed event listeners, to avoid null refs and CPU usage.
+        /// </summary>
         private void RemoveEventListeners()
         {
-            IEventService eventService = ServiceLocator.Get<IEventService>();
+            IEventService eventService = ServiceLocator.GetChecked<IEventService>();
+
             foreach (EventHandle eventHandle in _eventHandles)
             {
-                eventService?.RemoveListener(eventHandle);
-            } 
+                eventService.RemoveListener(eventHandle);
+            }
         }
 
-        #endregion
-
-        #region EVENT_CALLS
-        private void OnPreUpdate(ref EventContext context, in LastPreUpdateEvent e) { HandlePreUpdate(e.DeltaTime); }
```

Comment:
why do these event calls are removed ? Can you detail that a bit ?

Reply:
The automatic event call subscription was removed due to it causing wasted performance when the method was not overriden, so all Actors would have OnUpdate being called all the time even if it has no logic occurring.